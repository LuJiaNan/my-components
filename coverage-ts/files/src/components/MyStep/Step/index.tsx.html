
  <!DOCTYPE html>
  <html>
    <head>
      <title>index.tsx</title>
      <link href="https://cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css" type="text/css" rel="stylesheet">
      <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.js" type="text/javascript" charset="utf-8"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/mode/javascript/javascript.min.js" type="text/javascript" charset="utf-8"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.css" type="text/css" rel="stylesheet">
<script src="..\..\..\..\..\assets\source-file.js" type="text/javascript" charset="utf-8"></script>
<link href="..\..\..\..\..\assets\source-file.css" type="text/css" rel="stylesheet">
    </head>
    <body>
    <div style="margin-top:3em" class="ui container"><h1 class="ui header"><a href="..\..\..\..\..\index.html">TypeScript coverage report</a></h1><table style="margin-top:2em" class="ui celled table"><thead class=""><tr class=""><th class="">Filename</th><th class="">Percent</th><th class="">Threshold</th><th class="">Total</th><th class="">Covered</th><th class="">Uncovered</th></tr></thead><tbody class=""><tr class="positive"><td class="">src\components\MyStep\Step\index.tsx</td><td class="">83.00%</td><td class="">80%</td><td class="">100</td><td class="">83</td><td class="">17</td></tr></tbody></table><textarea id="editor" readonly="" style="margin-top:3em">import * as React from &quot;react&quot;;
import { useState, useEffect } from &quot;react&quot;;
import { Steps } from &quot;antd&quot;;
import StepProps from &quot;../index&quot;;

const defaultProps: StepProps = {
  steps: [],
  location: {
    pathname: &quot;&quot;
  },
  history: {
    push() {}
  }
};

const ref: any = React.createRef();

const Step: React.FC&lt;StepProps&gt; = props =&gt; {
  // 根据路由初始化步骤
  props = Object.assign({}, defaultProps, props);
  const currentStepRoute = props.location.pathname.slice(1);

  let currentStepNum = 0;
  props.steps.map((v, index) =&gt; {
    if (currentStepRoute.indexOf(v.path) &gt; -1) {
      currentStepNum = index + 1;
    }
  });
  const [step, goRoutes] = useState(currentStepNum);
  const getCurrentStep = () =&gt; step;

  const currentIndex = step - 1;
  const { steps } = props;
  const renderDom = props.steps[currentIndex];
  useEffect(() =&gt; {
    props.history.push(steps[currentIndex].path);
  }, [step]);

  return (
    &lt;React.Fragment&gt;
      &lt;Steps current={currentIndex}&gt;
        {steps.map((item: { text: string; description: string }) =&gt; (
          &lt;Steps.Step
            key={item.text}
            title={item.text}
            description={item.description}
          /&gt;
        ))}
      &lt;/Steps&gt;

      {React.createElement(renderDom.component, {
        ...props,
        ref,
        goRoutes,
        getCurrentStep
      })}
    &lt;/React.Fragment&gt;
  );
};

Step.defaultProps = defaultProps;

export default Step;
</textarea><pre id="annotations" style="display:none">[{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:15,&quot;character&quot;:6,&quot;text&quot;:&quot;ref&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:20,&quot;character&quot;:8,&quot;text&quot;:&quot;currentStepRoute&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:20,&quot;character&quot;:33,&quot;text&quot;:&quot;location&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:20,&quot;character&quot;:42,&quot;text&quot;:&quot;pathname&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:20,&quot;character&quot;:51,&quot;text&quot;:&quot;slice&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:23,&quot;character&quot;:19,&quot;text&quot;:&quot;v&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:24,&quot;character&quot;:8,&quot;text&quot;:&quot;currentStepRoute&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:24,&quot;character&quot;:25,&quot;text&quot;:&quot;indexOf&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:24,&quot;character&quot;:33,&quot;text&quot;:&quot;v&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:24,&quot;character&quot;:35,&quot;text&quot;:&quot;path&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:33,&quot;character&quot;:8,&quot;text&quot;:&quot;renderDom&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:35,&quot;character&quot;:10,&quot;text&quot;:&quot;history&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:35,&quot;character&quot;:18,&quot;text&quot;:&quot;push&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:35,&quot;character&quot;:43,&quot;text&quot;:&quot;path&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:50,&quot;character&quot;:27,&quot;text&quot;:&quot;renderDom&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:50,&quot;character&quot;:37,&quot;text&quot;:&quot;component&quot;},{&quot;file&quot;:&quot;src\\components\\MyStep\\Step\\index.tsx&quot;,&quot;line&quot;:52,&quot;character&quot;:8,&quot;text&quot;:&quot;ref&quot;}]</pre></div>
    <p class="footer-text">TypeScript Coverage Report generated by <a href="https://github.com/plantain-00/type-coverage">type-coverage</a> and <a href="https://github.com/alexcanessa/typescript-coverage-report">typescript-coverage-report</a> at Mon, 10 Aug 2020 02:25:51 GMT</p>
    </body>
  </html>
  