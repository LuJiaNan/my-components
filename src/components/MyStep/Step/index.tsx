import * as React from "react";
import { useState, useEffect } from "react";
import { Steps } from "antd";
import StepProps from "../index";

const defaultProps: StepProps = {
  steps: [],
  location: {
    pathname: ""
  },
  history: {
    push() {}
  }
};

const ref: any = React.createRef();

const Step: React.FC<StepProps> = props => {
  // 根据路由初始化步骤
  props = Object.assign({}, defaultProps, props);
  const currentStepRoute = props.location.pathname.slice(1);

  let currentStepNum = 0;
  props.steps.map((v, index) => {
    if (currentStepRoute.indexOf(v.path) > -1) {
      currentStepNum = index + 1;
    }
  });
  const [step, goRoutes] = useState(currentStepNum);
  const getCurrentStep = () => step;

  const currentIndex = step - 1;
  const { steps } = props;
  const renderDom = props.steps[currentIndex];
  useEffect(() => {
    props.history.push(steps[currentIndex].path);
  }, [step]);

  return (
    <React.Fragment>
      <Steps current={currentIndex}>
        {steps.map((item: { text: string; description: string }) => (
          <Steps.Step
            key={item.text}
            title={item.text}
            description={item.description}
          />
        ))}
      </Steps>

      {React.createElement(renderDom.component, {
        ...props,
        ref,
        goRoutes,
        getCurrentStep
      })}
    </React.Fragment>
  );
};

Step.defaultProps = defaultProps;

export default Step;
