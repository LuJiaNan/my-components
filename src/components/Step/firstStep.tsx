import * as React from 'react'
function Index() {
  return (    
    <div>我是step1</div>
  );
}

export default Index;
