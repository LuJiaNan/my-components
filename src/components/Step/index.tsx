import * as React from "react";
import { useState, useEffect } from "react";
import { Button, Steps } from "antd";

interface ISettingProps{
  steps: any[];
  history?: any;
  location?: any;
  showCancel?: boolean;
  cancelText?: string;
  finishText?: string;
  backPath?: string;
  showFinalLastStep?: boolean;
  finalSubmitFunctionName?: string;
}

const defaultProps: ISettingProps = {
  steps: [],
  cancelText: '取消',
  finishText: '完成',
  showFinalLastStep: true,
  showCancel: true,
};

const ref: any = React.createRef();

const Step: React.FC<ISettingProps> = (props) => {
  // 根据路由初始化步骤
  props = Object.assign({}, defaultProps, props);
  const currentStepRoute = props.location.pathname.slice(1);

  let currentStepNum = 0;
  props.steps.map((v,index) => {
    if(currentStepRoute.indexOf(v.path) > -1){
      currentStepNum = index + 1
    }
  })
  const [step, goRoutes] = useState(currentStepNum);

  const currentIndex = step - 1;
  const { steps } = props;
  const renderDom = props.steps[currentIndex];

  useEffect(() => {
    props.history.push(steps[currentIndex].path);
  }, [step]);
  return (
    <React.Fragment>
      <Steps current={currentIndex}>
        {steps.map((item: { text: string; description: string }) => (
          <Steps.Step
            key={item.text}
            title={item.text}
            description={item.description}
          />
        ))}
      </Steps>

      <renderDom.component ref={ref} {...props} goRoutes={goRoutes} />

      <Button.Group>
        {(() => {
          switch (step) {
            case 1: return ''
            case steps.length: return props.showFinalLastStep ? <Button onClick={() => goRoutes(step - 1)}>{"上一步"}</Button> : ''
            default: return <Button onClick={() => goRoutes(step - 1)}>{"上一步"}</Button>
          }
        })()}
        {step !== steps.length ? (
          <Button onClick={() => ref.current.onSubmit("handleSubmit")}>
            {"下一步"}
          </Button>
        ) : (
          ""
        )}
        {(step !== steps.length && props.showCancel) ? (
          <Button onClick={() => props.history.push(props.backPath || "/")}>
            {props.cancelText}
          </Button>
        ) : (
          ""
        )}
        {step === steps.length ? (
          <Button
            onClick={() =>
              props.finalSubmitFunctionName
                ? ref.current && ref.current[props.finalSubmitFunctionName]()
                : ref.current.onSubmit("handleSubmit")
            }
          >
            {props.finishText}
          </Button>
        ) : (
          ""
        )}
      </Button.Group>
    </React.Fragment>
  );
};

Step.defaultProps = defaultProps;

export default Step;
