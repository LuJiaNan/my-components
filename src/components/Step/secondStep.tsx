import * as React from 'react'
function Index() {
  return (    
    <div>我是step2</div>
  );
}

export default Index;
