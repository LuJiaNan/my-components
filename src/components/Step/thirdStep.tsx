import * as React from 'react'
function Index() {
  return (    
    <div>我是step3</div>
  );
}

export default Index;
