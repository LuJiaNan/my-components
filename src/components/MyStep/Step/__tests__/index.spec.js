import * as React from "react";
import { shallow } from "enzyme";
// import { render, fireEvent } from '@testing-library/react';
// import '@testing-library/jest-dom/extend-expect';
import { renderHook, act } from "@testing-library/react-hooks";
import Step from "../index.tsx";

function setup(props, options) {
  const defaultProps = {
    steps: [
      {
        text: "第一步",
        description: "我是第一步",
        path: "first",
        component: "div"
      },
      {
        text: "第二步",
        description: "我是第二步",
        path: "second",
        component: "div"
      }
    ],
    location: {
      pathname: "/first"
    },
    history: {
      push: jest.fn()
    }
  };

  const wrapper = shallow(
    <Step {...Object.assign({}, defaultProps, props)} />,
    options
  );
  return {
    wrapper
  };
}

describe("快照测试", () => {
  const props = {};
  it("step 组件快照", () => {
    const { wrapper } = setup(props);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("全方法覆盖", () => {
  // const props = {
  //   steps: [
  //     {
  //       text: "第一步",
  //       description: "我是第一步",
  //       path: "first",
  //       component: "div"
  //     },
  //     {
  //       text: "第二步",
  //       description: "我是第二步",
  //       path: "second",
  //       component: "div"
  //     }
  //   ],
  //   location: {
  //     pathname: "/first"
  //   },
  //   history: {
  //     push: jest.fn((a) => {})
  //   }
  // };

  // it("useEffect hook 测试", async () => {
  //   // jest.spyOn(React, 'useEffect').mockImplementation(f => f());
  //   const { findByTestId } = render(<Step {...props}/>);
  //   const $currentIndex = await findByTestId('currentIndex');
  //   expect($currentIndex).toHaveTextContent(props.steps[0].text);
  //   expect($currentIndex).toHaveTextContent(props.steps[0].description);
  //   expect($currentIndex).toHaveTextContent(props.steps[1].text);
  //   expect($currentIndex).toHaveTextContent(props.steps[1].description);
  // });

  const props = {}
  it("useEffect hook 测试", () => {
    const { wrapper } = setup(props)
    console.log(wrapper.debug())
    const { result, rerender } = renderHook(Step(props));
    console.log(result.current)
  })
});
