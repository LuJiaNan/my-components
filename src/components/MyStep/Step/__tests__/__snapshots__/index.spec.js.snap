// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 step 组件快照 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Step
    history={
        Object {
            "push": [Function],
          }
    }
    location={
        Object {
            "pathname": "/first",
          }
    }
    steps={
        Array [
            Object {
              "component": "div",
              "description": "我是第一步",
              "path": "first",
              "text": "第一步",
            },
            Object {
              "component": "div",
              "description": "我是第二步",
              "path": "second",
              "text": "第二步",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Steps
          current={0}
>
          <Step
                    description="我是第一步"
                    title="第一步"
          />
          <Step
                    description="我是第二步"
                    title="第二步"
          />
</Steps>,
        <div
          getCurrentStep={[Function]}
          goRoutes={[Function]}
          history={
                    Object {
                              "push": [Function],
                            }
          }
          location={
                    Object {
                              "pathname": "/first",
                            }
          }
          steps={
                    Array [
                              Object {
                                "component": "div",
                                "description": "我是第一步",
                                "path": "first",
                                "text": "第一步",
                              },
                              Object {
                                "component": "div",
                                "description": "我是第二步",
                                "path": "second",
                                "text": "第二步",
                              },
                            ]
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Step
              description="我是第一步"
              title="第一步"
/>,
            <Step
              description="我是第二步"
              title="第二步"
/>,
          ],
          "current": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "第一步",
            "nodeType": "class",
            "props": Object {
              "description": "我是第一步",
              "title": "第一步",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "第二步",
            "nodeType": "class",
            "props": Object {
              "description": "我是第二步",
              "title": "第二步",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "getCurrentStep": [Function],
          "goRoutes": [Function],
          "history": Object {
            "push": [Function],
          },
          "location": Object {
            "pathname": "/first",
          },
          "steps": Array [
            Object {
              "component": "div",
              "description": "我是第一步",
              "path": "first",
              "text": "第一步",
            },
            Object {
              "component": "div",
              "description": "我是第二步",
              "path": "second",
              "text": "第二步",
            },
          ],
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Steps
            current={0}
>
            <Step
                        description="我是第一步"
                        title="第一步"
            />
            <Step
                        description="我是第二步"
                        title="第二步"
            />
</Steps>,
          <div
            getCurrentStep={[Function]}
            goRoutes={[Function]}
            history={
                        Object {
                                    "push": [Function],
                                  }
            }
            location={
                        Object {
                                    "pathname": "/first",
                                  }
            }
            steps={
                        Array [
                                    Object {
                                      "component": "div",
                                      "description": "我是第一步",
                                      "path": "first",
                                      "text": "第一步",
                                    },
                                    Object {
                                      "component": "div",
                                      "description": "我是第二步",
                                      "path": "second",
                                      "text": "第二步",
                                    },
                                  ]
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Step
                description="我是第一步"
                title="第一步"
/>,
              <Step
                description="我是第二步"
                title="第二步"
/>,
            ],
            "current": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "第一步",
              "nodeType": "class",
              "props": Object {
                "description": "我是第一步",
                "title": "第一步",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "第二步",
              "nodeType": "class",
              "props": Object {
                "description": "我是第二步",
                "title": "第二步",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "getCurrentStep": [Function],
            "goRoutes": [Function],
            "history": Object {
              "push": [Function],
            },
            "location": Object {
              "pathname": "/first",
            },
            "steps": Array [
              Object {
                "component": "div",
                "description": "我是第一步",
                "path": "first",
                "text": "第一步",
              },
              Object {
                "component": "div",
                "description": "我是第二步",
                "path": "second",
                "text": "第二步",
              },
            ],
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
